typha/pkg/syncserver/sync_server.go

```diff
@@ -424,7 +424,9 @@ func (s *Server) serve(cxt context.Context) {
 			// Doing TLS, we must do the handshake...
 			tlsConn := conn.(*tls.Conn)
 			logCxt.Debug("TLS connection")
-			err = tlsConn.Handshake()
+			handshakeCxt, handshakeCancel := context.WithTimeout(cxt, s.config.PingInterval)
```

Comment:
Good catch, but I don't think PingInterval is the right one to use (it can be quite short and it's not a timeout).  We should really have a new timeout for this, but perhaps the WriteTimeout would be acceptable for a quick fix.

Reply:
Hi @fasaxc , thanks for your comment, I am more than happy to change it by the `WriteTimeout` if you believe this is the right path for a quick fix, my assumption when deciding by the `PingInterval` was based on the default value of it, which is `10s`, that initially sounded reasonable to me, because a TLS handshake is supposed to be pretty fast under normal operation, like under milliseconds, isn't it? The `WriteTimeout` has a default value of 120s, which seems quite a lot, meaning that a misbehaving client can still block the server up to 2 minutes, affecting the scalability of the Calico stack. 
I would appreciate your guidance on how we should proceed. Thanks in advance.

Reply:
I was about to say "don't worry, this is on its own goroutine" but now I look at it, that's not the case, looks like we're doing the TLS handshake synchronously after the `Accept()`, which is a mistake.

Let's just add  a new timeout, that's not too hard to do if you're up for it.  Jus tneed to add it in config_params.go and copy it "down the stack" in a couple of places.  Here's a commit that does it: https://github.com/projectcalico/calico/commit/f145093a8469ff7ac5250bfe80c43adaf95aef73

Reply:
@fasaxc , I have addressed your comment and added a new timeout to the config, I have also provided a PR to the documentation repository, please take a look.

Reply:
Sorry for not getting back sooner, yes, 10s seems fine to me.  I'd like it to be 2s but it's not easy to configure these values when using the operator to install Calico so let's compromise on 10s for now.